pkg/schema/schema.go

```diff
@@ -5,23 +5,43 @@ package schema
 
 import (
 	"fmt"
-
 	"github.com/k14s/ytt/pkg/template"
 	"github.com/k14s/ytt/pkg/yamlmeta"
 )
 
 type AnySchema struct {
 }
 
+type NullSchema struct {
+}
+
 type DocumentSchema struct {
-	Name    string
-	Source  *yamlmeta.Document
-	Allowed *DocumentType
+	Name       string
+	Source     *yamlmeta.Document
```

Comment:
These don't seem to be read anywhere. Is there a future requirement we're setting ourselves up for?

Reply:
There is a near term future requirement where we want to report error messages relating to schema type check. And for that to work we will need to have a reference to the schema source and the name field.